src/history/hs_rec.c

```diff
@@ -794,10 +817,23 @@ __wt_hs_delete_key(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_t btre
         WT_ERR(hs_cursor->get_key(hs_cursor, &hs_btree_id, &hs_key, &hs_start_ts, &hs_counter));
         ++hs_counter;
     }
-
-    WT_ERR(__hs_delete_reinsert_from_pos(session, hs_cursor, btree_id, key, WT_TS_NONE, reinsert,
-      true, error_on_ts_ordering, &hs_counter, NULL));
-
+    /* Get the history store cursor position */
+    WT_ERR_NOTFOUND_OK(__hs_cursor_pos(session, hs_cursor, btree_id, key, WT_TS_NONE,
+                         error_on_ts_ordering, NULL, &non_ts_updates),
+      true);
+
+    /* 
+     * If we have non-timestamped updates, skip the reinserts as we do not need to zero the
+     * timestamps already at zero. If we have non-timestamped deletes (then reinsert is false)
+     * continue into the function to delete only. If we do not have a valid cursor position then
+     * nothing is found, return.
+     */
+    if (!(reinsert && non_ts_updates) && ret == 0) {
```

Comment:
same here, no need for brackets

Reply:
removed here [1c4f7bb](https://github.com/wiredtiger/wiredtiger/pull/10686/commits/1c4f7bb287446e1166c6d19da2501a9c365bff50)